Composer/packages/client/src/pages/language-understanding/index.tsx

```diff
@@ -154,33 +126,19 @@ const LUPage: React.FC<DefineConversationProps> = props => {
       </div>
       <div css={ContentStyle} data-testid="LUEditor">
         <div css={projectContainer}>
-          <Tree variant="large" css={projectTree}>
-            <div css={projectWrapper}>
-              <Nav
-                onLinkClick={(ev, item) => {
-                  item && onSelect(item.id);
-                  ev && ev.preventDefault();
-                }}
-                styles={{
-                  root: {
-                    /* override dulplicate selected mark bellow All*/
-                    selectors: {
-                      'ul>li>ul button.ms-Nav-chevronButton:after': {
-                        borderLeft: 'none',
-                      },
-                    },
-                  },
-                  chevronButton: {
-                    backgroundColor: 'transparent',
-                  },
+          {navLinks.map(dialog => {
+            return (
+              <div
+                css={dialogItem(activeDialog ? activeDialog.id === dialog.id : dialog.id === '_all')}
+                key={dialog.id}
+                onClick={() => {
+                  onSelect(dialog.id);
```

Comment:
duplicate logic with LG. You need a shared component, `<NavLinks />`

Reply:
`LU` and `LG` page could be treated as containers. They just transform `dialogs` to `navLinks`, so transformation logic lives here.

Rendering work could be delegated to a new shared component `<NavLinks navLinks={[.....]} />`.

Reply:
It has some differences between LG and LU page.
1. LG page nav tree root is `common.lg` file 
2. LU page nav tree root is a virtual `merged *.lu` file.
we can't unified this.

The rest (dialog tree) in LG/LU is same, we could wrap it as a component.